### PR TITLE
Added tape mode

### LINKED ISSUE #2858: Add single line mode

### PR DESCRIPTION
**Pull request by Miodec**
Mode which shows one line and scrolls horizontally

Closes #2858

### CODE REVIEW

**Comment by Ferotiq** [comment]
To fix: with tape mode off, the top line is always the typing line, although it should go to the middle line first and stay on the middle line.

**Comment by Miodec** [comment]
> To fix: with tape mode off, the top line is always the typing line, although it should go to the middle line first and stay on the middle line.

Fixed

File: frontend/src/scripts/test/caret.ts [hunk 1]
```diff
@@ -78,18 +78,43 @@ export async function updatePosition(): Promise<void> {
...
-        ($(currentLetter).width() as number) -
-        (caret.width() as number) / 2
-      : currentLetterPosLeft -
-        ($(currentLetter).width() as number) +
+
+  if (Config.tapeMode === "letter") {
+    newLeft =
+      ($(<HTMLElement>document.querySelector("#wordsWrapper")).width() ?? 0) /
+        2 -
+      (caret.width() as number) / 2;
```
Comment (Ferotiq) [inline frontend/src/scripts/test/caret.ts:86]: you probably should store caret width as a variable

File: frontend/src/scripts/test/test-ui.ts [hunk 2]
```diff
@@ -547,6 +617,7 @@ export function lineJump(currentTop: number): void {
     }
   }
   currentTestLine++;
+  // }
```
Comment (Ferotiq) [inline frontend/src/scripts/test/test-ui.ts:620]: delete

**Comment by Miodec** [comment]
@Ferotiq fixed both issues

**Review by Ferotiq — APPROVED** [review]
LGTM IIW

**Pull request merged**